File: src/main/scala/org/roadlessforest/osm/config/ConfigurationFactory.scala
package org.roadlessforest.osm.config

import java.util.Properties

import org.apache.hadoop.conf.Configuration

import scala.collection.JavaConversions._

/**
 * Just a place to keep the config in one place
 *
 * Created by tempehu on 22-Dec-14.
 */
object ConfigurationFactory {

  def get: Configuration = {

    val props: Properties = getProperties("hbase-config.properties")
    val conf = new Configuration
    props.keys().foreach(f => conf.set(f.toString, props.get(f.toString).toString))
    conf
  }

  def getPrecedence: Map[Int, Int] = {

    val props: Properties = getProperties("raster-priority.properties")

    props.toMap.map(f => f._1.toInt -> f._2.toInt)

  }

  def getProperties(propertiesFileName: String): Properties = {
    val props = new Properties()

    val loader = Thread.currentThread().getContextClassLoader

    val resourceStream = loader.getResourceAsStream(propertiesFileName)

    props.load(resourceStream)

    props
  }
}

File: src/main/java/org/roadlessforest/xyz/ConfigurationFactory.java
package org.roadlessforest.xyz;

import org.apache.hadoop.conf.Configuration;

import java.io.IOException;
import java.io.InputStream;
import java.util.Enumeration;
import java.util.Properties;

import static org.osgeo.proj4j.parser.Proj4Keyword.f;

/**
 * Created by willtemperley@gmail.com on 13-Mar-17.
 */
public class ConfigurationFactory {

    public static Properties getProperties(String propertiesFileName) throws IOException {
        Properties props = new Properties();
        ClassLoader loader = Thread.currentThread().getContextClassLoader();
        InputStream resourceStream = loader.getResourceAsStream(propertiesFileName);
        props.load(resourceStream);
        return props;
    }

    public static Properties hbaseProperties() throws IOException {
        return getProperties("hbase-config.properties");
    }


    public static Configuration getConfiguration() throws IOException {

        Configuration configuration = new Configuration();
        Properties properties = hbaseProperties();
        Enumeration<?> enumeration = properties.propertyNames();

        while (enumeration.hasMoreElements()) {
            Object key = enumeration.nextElement();
            Object o = properties.get(key);
            configuration.set(key.toString(), o.toString());
        }

        return configuration;
    }

    public static void main(String[] args) throws IOException {
        Configuration configuration = getConfiguration();
        System.out.println("configuration = " + configuration);
    }
}
